include/api.php

```diff
@@ -3629,6 +3629,80 @@ function api_direct_messages_destroy($type)
 /// @TODO move to top of file or somewhere better
 api_register_func('api/direct_messages/destroy', 'api_direct_messages_destroy', true, API_METHOD_DELETE);
 
+/**
+ * Unfollow Contact
+ *
+ * @brief unfollow contact 
+ *
+ * @param string $type Known types are 'atom', 'rss', 'xml' and 'json'
+ * @return string|array
+ * @see https://developer.twitter.com/en/docs/accounts-and-users/follow-search-get-users/api-reference/post-friendships-destroy.html
+ */
+function api_friendships_destroy($type)
+{
+	$a = api_user();
+
+	if ($a === false) {
+		throw new ForbiddenException();
+	}
+
+	$uid = local_user();
+
+	$contact_id = defaults($_REQUEST, 'user_id');
+
+	if ($contact_id == null) {
+		logger("No user_id specified", LOGGER_DEBUG);
+		throw new BadRequestException("no user_id specified");
+	}
+
+	// Get Contact by given id
+	$contact = DBA::selectFirst('contact', ['url'], ['id' => $contact_id, 'uid' => 0, 'self' => false]);
+
+	if(!DBA::isResult($contact)) {
+		logger("No contact found for ID" . $contact_id, LOGGER_DEBUG);
+		throw new NotFoundException("no contact found to given ID");
+	}
+
+	$url = $contact["url"];
+
+	$condition = ["`uid` = ? AND (`rel` = ? OR `rel` = ?) AND (`nurl` = ? OR `alias` = ? OR `alias` = ?)",
+			$uid, Contact::SHARING, Contact::FRIEND, normalise_link($url),
+			normalise_link($url), $url];
+	$contact = DBA::selectFirst('contact', [], $condition);
+
+	if (!DBA::isResult($contact)) {
+		logger("Not following Contact", LOGGER_DEBUG);
+		throw new NotFoundException("Not following Contact");
+	}
+
+	if (!in_array($contact['network'], Protocol::NATIVE_SUPPORT)) {
+		logger("Not supported", LOGGER_DEBUG);
+		throw new ExpectationFailedException("Not supported");
+	}
+
+	$dissolve = ($contact['rel'] == Contact::SHARING);
+
+	$owner = User::getOwnerDataById($uid);
+	if ($owner) {
+		Contact::terminateFriendship($owner, $contact, $dissolve);
+	}
+	else {
+		logger("No owner found", LOGGER_DEBUG);
+		throw new NotFoundException("Error Processing Request");
+	}
+
+	// Sharing-only contacts get deleted as there no relationship any more
+	if ($dissolve) {
+		Contact::remove($contact['id']);
+	} else {
+		DBA::update('contact', ['rel' => Contact::FOLLOWER], ['id' => $contact['id']]);
+	}
+
+	$answer = ['result' => 'ok', 'user_id' => $contact_id, 'contact' => 'contact deleted'];
+	return api_format_data("friendships-destroy", $type, ['result' => $answer]);
```

Comment:
Yes, but no the response does not fit. 
I'm not getting the Twitter API response. It says https://developer.twitter.com/en/docs/accounts-and-users/follow-search-get-users/api-reference/post-friendships-destroy.html

    {user-object,
      "status": {tweet-object}
    }

But is this valid json ? I thought it always needs an "identifier like "status" : "value".

    {
    json_encode($contact),
     "status": ?
    }

Could you give me a hit ?

Reply:
The return value is actually described above:

> Returns the unfollowed user when successful. Returns a string describing the failure condition when unsuccessful.

So the return should be the user JSON object  when it was successful, an error string otherwise. The example response doesn't seem to match the function.

Reply:
What about this?